### PR TITLE
cisco secure email connector

### PR DESCRIPTION
**Pull request by thangaraj-ramesh**
cisco secure email connector

### CODE REVIEW

File: stix_shifter_modules/cisco_secure_email/configuration/lang_en.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            "description": "More details on the data source setting can be found in the specified link"
+        },
+        "selfSignedCert": {
+            "label": "Cisco Secure Email HTTPS connection certificate",
+            "description": "Use self-signed Security Sockets Layer (SSL) or CA certificate for HTTPS services"
+        },
+         "options": {
+             "result_limit": {
+                 "label": "Result size limit",
+                 "description": "The maximum number of entries or objects that are returned by search query.  Valid input range is {{min}} to {{max}}."
```
Comment (mdazam1942) [inline stix_shifter_modules/cisco_secure_email/configuration/lang_en.json:22]: do you see {{min}} and {{max}} showing up in the UI description?

Comment (thangaraj-ramesh) [reply stix_shifter_modules/cisco_secure_email/configuration/lang_en.json:22]: In the UI, it shows from the common configuration that values are not taken from the connector configuration. 

The below description is shown in CP4S
"The maximum number of entries or objects that are returned by search query. Valid input range is 1 to 10000000."

File: stix_shifter_modules/cisco_secure_email/stix_transmission/error_mapper.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+error_mapping = {
+    100: ErrorCode.TRANSMISSION_INVALID_PARAMETER,
+    406: ErrorCode.TRANSMISSION_CONNECT,
+    408: ErrorCode.TRANSMISSION_CONNECT,
+    503: ErrorCode.TRANSMISSION_CONNECT,
+    400: ErrorCode.TRANSMISSION_QUERY_PARSING_ERROR,
+    401: ErrorCode.TRANSMISSION_AUTH_CREDENTIALS,
+    403: ErrorCode.TRANSMISSION_FORBIDDEN,
+    404: ErrorCode.TRANSMISSION_SEARCH_DOES_NOT_EXISTS,
```
Comment (mdazam1942) [inline stix_shifter_modules/cisco_secure_email/stix_transmission/error_mapper.py:13]: i think this should be invalid query. `no_results` error code is not appropriate. this error code usually returns when query syntax is invalid

File: stix_shifter_modules/cisco_secure_email/stix_transmission/connector.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                return_obj = self.exception_response(response_code, response_dict.get('errorSummary', ''))
+
+        except Exception as ex:
+            if "server timeout_error" in str(ex):
+                response_dict['code'] = 503
+            elif "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error while pinging: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
```
Comment (mdazam1942) [inline stix_shifter_modules/cisco_secure_email/stix_transmission/connector.py:129]: you need to handle SSLError as well. i have got an error with `unknown` error code when i put invalid certificate.

```
{
    "success": false,
    "connector": "cisco_secure_email",
    "error": "cisco_secure_email connector error => [X509] PEM lib (_ssl.c:<code>)",
    "code": "unknown"
}
```

File: stix_shifter_modules/cisco_secure_email/stix_transmission/error_mapper.py (hunk 2 above)
Comment (thangaraj-ramesh) [reply stix_shifter_modules/cisco_secure_email/stix_transmission/error_mapper.py]: Updated error code as invalid_query

File: stix_shifter_modules/cisco_secure_email/stix_transmission/connector.py (hunk 3 above)
Comment (thangaraj-ramesh) [reply stix_shifter_modules/cisco_secure_email/stix_transmission/connector.py:137]: Updated code to handle Invalid certificate exception.

**Review by mdazam1942 — APPROVED** [review]

**Pull request merged**